File: Java/SearchingAlgorithms/LinearSearch.java
package searching_algo;
import java.util.Scanner;
public class LinearSearch {
    public static  void Search(int[] arr,int size,int val){
        boolean flag=false;
        
        for(int i=0;i<size;i++)
        {
            if (arr[i]==val){
                System.err.println("element is at "+i+" index");
                flag=true;
            }
        }
        if (!flag){
            System.out.println("element not found");
        }

        
    }



    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);

        System.out.println("enter the size of the Array");

        int size=sc.nextInt();

        int[] arr=new int[size];

        for (int i=0;i<size;i++){
            arr[i]=sc.nextInt();
        }

        while(true){
            System.out.println("1.enter element to search\n 2.quit\n");
            int choice=sc.nextInt();
            if (choice==1){
                System.out.println("enter value to search\n");
                int val=sc.nextInt();
                Search(arr,size, val);
            }
            else{
                break;
            }
        }
        sc.close();

        
    }
}

File: Python/SearchingAlgorithms/linearSearch.py
arr=list(map(int,input('enter space seperated values\n').split()))

def linearSearch(arr):
        flag=False
        val=int(input("enter value to search"))
        for i in range(len(arr)):
            if arr[i]==val:
                print(f'{val} is in {i} index')
                flag=True
        if not flag:
            print('element not exists')

while True:
    choice=int(input("1.search\n2.quit\n"))
    if choice==1:
         linearSearch(arr)

    else:
        break
